src/backend/db/feature-permissions.js

```diff
@@ -0,0 +1,219 @@
+const {google} = require('googleapis');
+const { OAuth2Client } = require('google-auth-library');
+
+/**
+ * Translate Group Keys to Group Ids
+ * 
+ * @param {*} db: The database interface object 
+ * @param {Array[String]} groupKeys 
+ * 
+ * @returns Array of Group Ids
+ */
+const keysToIds = async (db, groupKeys) => {
+  let groupIds = [];
+  if (groupKeys && groupKeys.length > 0) {
+    try {
+      let query = 'SELECT id FROM groups WHERE group_key IN (' + groupKeys.map(_ => '?').join(',') + ');'
+      groupIds = ( await db.raw(query, [...groupKeys] )).rows.map(row => row.id);
+    } catch(e) {
+      console.log(e);
+      groupIds = [];
+    }
+  } 
+  return groupIds;
+}
+
+/**
+ * Return Group Ids and Group Keys, given a set of Group Ids as input
+ * 
+ * @param {*} db: The database interface object 
+ * @param {Array[String]} groupIds
+ */
+const appendKeysToIds = async (db, groupIds) => {
+  // Translate Group Keys into Group IDs
+  let groupObjs = [];
+  if (groupIds && groupIds.length > 0) {
+    try {
+      // Reference: For array bindings syntax, please see: https://knexjs.org/#Raw-Bindings 
+      let query = 'SELECT id, group_key FROM groups WHERE id IN (' + groupIds.map(_ => '?').join(',') + ');';
+      groupObjs = ( await db.raw(query, [...groupIds] )).rows;
+    } catch(e) {
+      console.log(e);
+      groupObjs = [];
+    }
+  }
+  return groupObjs;
```

Comment:
It looks like we could combine this function appendKeysToIds  with the previous function keysToIds .